src/Planners/include/BipedalLocomotion/Planners/UnicyclePlanner.h

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file UnicyclePlanner.h
+ * @authors Diego Ferigo, Stefano Dafarra
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_UNICYCLE_PLANNER_H
+
+#include "BipedalLocomotion/Contacts/ContactList.h"
+#include "BipedalLocomotion/ParametersHandler/IParametersHandler.h"
+#include "BipedalLocomotion/System/Advanceable.h"
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <memory>
+
+namespace BipedalLocomotion::Planners
+{
+struct UnicycleKnot;
+class UnicyclePlanner;
+struct UnicyclePlannerInput;
+struct UnicyclePlannerOutput;
+} // namespace BipedalLocomotion::Planners
+
+struct BipedalLocomotion::Planners::UnicycleKnot
+{
+    UnicycleKnot(const double x = 0.0,
+                 const double y = 0.0,
+                 const double dx = 0.0,
+                 const double dy = 0.0,
+                 const double t = 0.0)
+        : x(x)
+        , y(y)
+        , dx(dx)
+        , dy(dy)
+        , time(t)
+    {
+    }
+
+    UnicycleKnot(const Eigen::Vector2d& position = {0, 0},
+                 const Eigen::Vector2d& velocity = {0, 0},
+                 const double time = 0.0)
+        : x(position[0])
+        , y(position[1])
+        , dx(velocity[0])
+        , dy(velocity[1])
+        , time(time)
+    {
+    }
+
+    bool operator==(const UnicycleKnot& rhs)
+    {
+        return this->x == rhs.x && this->y == rhs.y && this->dx == rhs.dx && this->dy == rhs.dy
+               && this->time == rhs.time;
+    }
+
+    double x; ///< The knot x coordinates.
+    double y; ///< The knot y coordinates.
+
+    double dx = 0.0; ///< The knot x velocity.
+    double dy = 0.0; ///< The knot y velocity.
+
+    double time = 0.0; ///< The knot activation time.
+};
+
+struct BipedalLocomotion::Planners::UnicyclePlannerInput
+{
+    UnicyclePlannerInput(const std::vector<UnicycleKnot>& knots,
+                         const double tf = 0.0,
+                         const double t0 = 0.0)
+        : t0(t0)
+        , tf(tf)
+        , knots(knots)
```

Comment:
Sure, note that this works as expected, but I agree on readability. In QtCreator the colors or arguments and class attributes are different and it looks ok there, but here it could be confusing.